lib/atmosphere.rb

```diff
@@ -97,8 +97,12 @@ def self.ability_class
     @@ability_class.constantize
   end
 
-  mattr_reader :nics
-  @@nics = {}
+  def self.nic_provider(compute_site)
+    class_name = compute_site.nic_provider_class_name
+    nic_provider_class = class_name ? class_name.constantize :
+      Atmosphere::NicProvider::NullNicProvider
+    nic_provider_class.new(compute_site.nic_provider_config)
```

Comment:
I already mentioned about it in plgrid PR but just to remember: Maybe instead of passing string configuration can be converted from json structure into `key:value` hash and passed as NIC provider configuration?